docs/reference/migrating-example.md

```diff
@@ -0,0 +1,171 @@
+
+3. Identify the project to be migrated first
+
+We decided to start with the [`gradle-client` subproject](https://github.com/gradle/gradle-client/tree/main/gradle-client), as this was the most important and fully featured project contained in the Gradle Client build, which would be the most interesting to investigate.
+For your own migrations, you should probably consider which projects would be _simplest_ to migrate.
+Projects that are simpler and most similar to our declarative samples are good candidates.
+If your project _doesn't_ already use Gradle's Kotlin DSL, you'll want to migrate to this first from your Groovy build using the existing [migration guide](https://docs.gradle.org/current/userguide/migrating_from_groovy_to_kotlin_dsl.html) in the Gradle documentation.
+
+3. Add the proper ecosystem plugin
+
+The Declarative Gradle prototype maintains a KMP prototype plugin.
+As the `gradle-client` project is a Kotlin Multiplatform project, we added this plugin to the project's existing `settings.gradle.kts` file.
+Also, be sure to include the `gradlePluginPortal()` as a repository if your project doesn't already include this.
+
+
+```
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+    }
+}
+
+plugins {
+    id("org.gradle.experimental.kmp-ecosystem").version("0.1.41")
+}
+
+```
+
+4. Use the existing prototype plugin's software types to migrate basics
+
+Thing like declaring typical `implementation`, `api` and testing dependencies should be already supported out-of-the-box with the existing Software Types provided by Gradle's prototype plugins.
+Migrating your dependencies to these is a good test of your setup.
+Remember that Declarative Software Types can be used from KTS files.
+
+We added the following to the `gradle-client` project's `build.gradle.kts` file:
+
+
+```
+kotlinApplication {
+    targets {
+        jvm {
+            jdkVersion = 17
+
+            dependencies {
+                implementation(project(":build-action"))
+                implementation(project(":mutations-demo"))
+
+                implementation(project.dependencies.platform(libs.kotlin.bom))
+                implementation(project.dependencies.platform(libs.kotlinx.coroutines.bom))
+                implementation(project.dependencies.platform(libs.kotlinx.serialization.bom))
+                implementation(project.dependencies.platform(libs.ktor.bom))
+
+                implementation(libs.gradle.tooling.api)
+
+                implementation(libs.sqldelight.extensions.coroutines)
+                implementation(libs.sqldelight.runtime)
+
+                // ...
+            }
+        }
+    }
+}
+```
+
+And we _removed_ these dependencies from the `jvmMain.dependencies` block where they were previously declared.
+Set the org.gradle.kotlin.dsl.dcl flag in the project's properties file.
```

Comment:
This looks out of place?

Reply:
I'll edit to make it stand out more.  It is needed as part of this step, else Software Types won't work in KTS files.

Reply:
I think this should be part of the "initial setup"